### PR TITLE
Treat warnings as errors in the "production" mode

### PR DESCRIPTION
**Pull request by martnpaneq**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix (tests): When running automated tests in the "production" mode, warnings will be treated as errors. It allows the detection of webpack messages that may impact a build but does not break the process.

Fix (tests): The manual and automated test environments do not report a warning regarding the missing `timers` module.

---

Closes cksource/ckeditor5-internal#3299.

---

### Additional information

*For example – encountered issues, assumptions you had to make, other affected tickets, etc.*

### CODE REVIEW

File: packages/ckeditor5-dev-tests/bin/testautomated.js [hunk 1]
```diff
@@ -13,6 +13,14 @@ const tests = require( '../lib/index' );
 
 const options = tests.parseArguments( process.argv.slice( 2 ) );
 
+// We want to mark tests as failed if any warnings are logged. See https://github.com/cksource/ckeditor5-internal/issues/3299
+if ( options.production ) {
+	console.warn = errors => {
+		console.log( errors );
```
Comment (martnpaneq) [inline packages/ckeditor5-dev-tests/bin/testautomated.js:19]: I tried calling `console.error` instead, but it does not break the execution.
The question is should we also throw error on `console.error`?

Comment (psmyrek) [reply packages/ckeditor5-dev-tests/bin/testautomated.js:19]: I think that both `console.warn` and `console.error` should break the execution and throw an error.

Comment (pomek) [reply packages/ckeditor5-dev-tests/bin/testautomated.js:19]: During a call, we created a custom plugin. Could you guys verify if it works on your machines too?

Webpack config for automated tests:

```js
if ( options.production ) {
	config.plugins.push(
		new class TreatWarningsAsErrorsWebpackPlugin {
			apply( compiler ) {
				compiler.hooks.shouldEmit.tap( 'TreatWarningsAsErrorsWebpackPlugin', compilation => {
					const stats = compilation.getStats();

					compilation.errors = [
						...compilation.errors,
						...compilation.warnings
					];

					compilation.warnings = [];

					if ( compilation.errors.length ) {
						return false;
					}
				} );
			}
		}
	)
}
```

Comment (psmyrek) [reply packages/ckeditor5-dev-tests/bin/testautomated.js:19]: It works. Test run is terminated.

**Comment by pomek** [comment]
The same error occurs for manual tests.

image

**Comment by pomek** [comment]
I checked if the polyfill is needed at all. It turns out that everything works as expected without it.

**Pull request merged**